### PR TITLE
Feature/test branch

### PR DESCRIPTION
**Pull request by SamanAtashi**
# TDD project
Features introduced are as follows:

- Write unit tests using Rspec.
- Understand an existing codebase in a way that allows for adding tests.
- Write Rspec unit tests with mocks.
- Understand the TDD concept.

### CODE REVIEW

File: tests/spec/solver_spec.rb [hunk 1]
```diff
@@ -0,0 +1,42 @@
...
+    end
+
+    it 'return 5040 when given 7' do
+      expect(solver.factorial(7)).to eq 5040
+    end
+
+    it 'return 720 when given 6' do
+      expect(solver.factorial(6)).to eq 720
+    end
+  end
```
Comment (omar-labana) [inline tests/spec/solver_spec.rb]: - [x] Please add a test case that handles negative value which raises an exception as an expectation, I'd suggest using `to raise_exception` if a negative value was given as an argument.

**Review by omar-labana — CHANGES_REQUESTED** [review]
Hi @SamanAtashi and @Lordkaito, 

Excellent work so far! 👌
You have implemented almost all the required features. ✔️ 
There are some cases that you need to adjust but you are almost done! 😀

## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._


Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Review by omar-labana — APPROVED** [review]
Hi @SamanAtashi and @Lordkaito , 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 



Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.


------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._
